examples/vision/learnable_resizer.py

```diff
@@ -0,0 +1,273 @@
+"""
+Title: Learning to Resize in Computer Vision
+Author: [Sayak Paul](https://twitter.com/RisingSayak)
+Date created: 2021/04/30
+Last modified: 2021/04/30
+Description: How to optimally learn representations of images for a given resolution.
+"""
+"""
+It is a common belief that if we constrain vision models to perceive things as humans do,
+their performance can be improved. For example, in [this
+work](https://arxiv.org/abs/1811.12231), Geirhos et. al showed that the vision models
+pre-trained on the ImageNet-1k dataset are biased toward texture whereas human beings
+mostly use the shape descriptor to develop a common perception. But does this belief
+always apply especially when it comes to improving the performance of vision models? 
+
+It turns out it may not always be the case. When training vision models, it is common to
+resize images to a lower dimension ((224x224), (299x299), etc.) to allow mini-batch
+learning and also to keep up the compute limitations.  We generally make use of image
+resizing methods like **bilinear interpolation** for this step and the resized images do
+not lose much of their perceptual character to the human eyes. In [Learning to Resize
+Images for Computer Vision Tasks](https://arxiv.org/abs/2103.09950v1), Talebi et. al show
+that if we try to optimize the perceptual quality of the images for the vision models
+rather than the human eyes, their performance can further be improved. They investigate
+the following question: 
+
+**For a given image resolution and a model, how to best resize the given images?**
+
+As shown in the paper, this idea helps to consistently improve the performance of the
+common vision models (pre-trained on ImageNet-1k) like DenseNet121, ResNet50,
+MobileNetV2, and EfficientNets. In this example, we will implement the learnable image
+resizing module as proposed in the paper and demonstrate that on the
+[Cast-Vs-Dogs dataset](https://www.microsoft.com/en-us/download/details.aspx?id=54765)
+using the [DenseNet121](https://arxiv.org/abs/1608.06993) model. 
```

Comment:
Very cool intro.

Suggestion:

- Break into more paragraphs for improved readability
- Add the first paper's name (ImageNet-trained CNNs are biased towards texture; increasing shape bias improves accuracy and robustness)
- "can be biased" <- "are biased" (nothing is 100% certain?)
- "DenseNet-121", "ResNet-50" with a dash
- "Kaggle Cats and Dogs" <- "Cast-Vs-Dogs" (typo)
- "DenseNet-121 architecture" <- "...model" (https://www.tensorflow.org/api_docs/python/tf/keras/applications/DenseNet121)
- Resolution (Number x Number)
- 80 char per line

```suggestion
It is a common belief that if we constrain vision models to perceive things as
humans do, their performance can be improved. For example, in
[ImageNet-trained CNNs are biased towards texture; increasing shape bias improves accuracy and robustness](https://arxiv.org/abs/1811.12231)
, Geirhos et al. showed that vision models pre-trained on the ImageNet 1000
dataset can be biased toward texture, whereas human beings mostly use the shape 
descriptor to develop a common perception. 

_However, does this belief always apply especially when it comes to improving
the performance of vision models?_ It turns out it may not always be the case.
When training vision models, it is common to resize images to a lower dimension
(such as (224 x 224), (299 x 299), etc.) to allow mini-batch learning and keep 
up with the compute limitations.

For this step, we generally use image resizing methods, such as **bilinear
interpolation**, while the resized images do not lose much of their perceptual
character to the human eyes. 

In [Learning to Resize Images for Computer Vision Tasks](https://arxiv.org/abs/2103.09950v1)
, Talebi et. al demonstrate that if we try to optimize the perceptual quality
of the images for the vision models rather than the human eyes, their
performance can further be improved. They investigate the following question: _for a given image resolution and a model, how can we best resize the given
images?_ As shown in the paper, this idea helps to consistently improve the
performance of the common vision models (pre-trained on ImageNet-1k), such as
DenseNet-121, ResNet-50, MobileNetV2, and EfficientNets. 

In this example, we will implement the learnable image resizing module as
proposed in the paper and demonstrate that on the
[Kaggle Cats and Dogs dataset](https://www.microsoft.com/en-us/download/details.aspx?id=54765)
using the [DenseNet-121](https://arxiv.org/abs/1608.06993) architecture.
```

Reply:
> 80 char per line

(Nit: as per @ebursztein's suggestion in another example by a different user, maybe we should have lines no longer than 80 characters (instead of 90 or 100). Maybe, this can be done your future example, which are awesome as always.)

Reply:
Thank you!

I try to keep the characters in a line such that they are visually not problematic.